pkg/proc/bininfo.go

```diff
@@ -149,21 +149,156 @@ func (err *ErrFunctionNotFound) Error() string {
 
 // FindFileLocation returns the PC for a given file:line.
 // Assumes that `file` is normalized to lower case and '/' on Windows.
-func FindFileLocation(p Process, fileName string, lineno int) ([]uint64, error) {
-	pcs, err := p.BinInfo().LineToPC(fileName, lineno)
-	if err != nil {
-		return nil, err
+func FindFileLocation(p Process, filename string, lineno int) ([]uint64, error) {
+	// A single file:line can appear in multiple concrete functions, because of
+	// generics instantiation as well as multiple inlined calls into other
+	// concrete functions.
+
+	bi := p.BinInfo()
+
+	fileFound := false
+	pcs := []line.PCStmt{}
+	for _, image := range bi.Images {
+		for _, cu := range image.compileUnits {
+			if cu.lineInfo == nil || cu.lineInfo.Lookup[filename] == nil {
+				continue
+			}
+
+			fileFound = true
+			pcs = append(pcs, cu.lineInfo.LineToPCs(filename, lineno)...)
+		}
+	}
+
+	if len(pcs) == 0 {
+		// Check if the line contained a call to a function that was inlined, in
+		// that case it's possible for the line itself to not appear in debug_line
+		// at all, but it will still be in debug_info as the call site for an
+		// inlined subroutine entry.
+		for _, pc := range bi.inlinedCallLines[fileLine{filename, lineno}] {
+			pcs = append(pcs, line.PCStmt{PC: pc, Stmt: true})
+		}
 	}
+
+	if len(pcs) == 0 {
+		return nil, &ErrCouldNotFindLine{fileFound, filename, lineno}
+	}
+
+	// 2. assign all occurences of filename:lineno to their containing function
```

Comment:
I don't see a `1.`, but there's a `2.` and `.3`, etc... in this comment list.

Reply:
Added.